### PR TITLE
fix(topology): resolve device ids instead of silently returning nothing

### PR DESCRIPTION
**Pull request by jpennin5**
## Description

**Problem:** a switch carrying nine hosts reported none, with no error.

**Root cause:** the topology is built from the Integration API, whose
device ids are UUIDs. Every other device tool in this server speaks the
legacy controller `_id` or takes a MAC. `get_port_mappings` and
`get_device_connections` filtered connections by exact string match on the
topology's node ids, so passing an id from any other tool matched nothing
and returned `{}` or `[]` — indistinguishable from a device that genuinely
has no connections. It is a lookup that answers "no" to a question it
never understood.

**Fix:** resolve the identifier against the node list by node id, MAC, or
name, and raise `ResourceNotFoundError` when it matches none of them — "I
do not know this device" and "this device has no connections" are
different answers and must not share a representation. `get_port_mappings`
reports the resolved node id as `device_id` and echoes the caller's
identifier as `requested_id`.

**Note on tests:** one existing test built connections referencing an
`uplinkDeviceId` with no matching device, which a real controller never
reports; its fixture now includes the switch, and its mock dispatches on
URL rather than call order, which the per-device enrichment pass had made
brittle.

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)

Callers already passing topology node ids are unaffected; unknown
identifiers now raise instead of returning an empty result.

## Testing

- Tests cover resolution by node id, MAC, and name, the
  `ResourceNotFoundError` path, and the `requested_id` echo.
- `pytest tests/unit/` and `pre-commit run --all-files` pass.
- `API.md` documents the accepted identifier forms and the error
  behavior.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (API.md, CHANGELOG.md)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published (none — stands alone)

## AI Assistance

This PR was created with assistance from Claude Code.

**Human Review Status:** ✅ Reviewed and approved by @jpennin5
**Testing:** Full unit suite passing; resolution behavior verified against
a live Network 10.5.67 controller.
**Security Review:** Completed — all fixture values are synthetic or from
documentation ranges.

### CODE REVIEW

File: src/tools/topology.py [hunk 1]
```diff
@@ -312,6 +313,32 @@ async def get_network_topology(
...
+    wanted = device_id.strip().lower()
+    nodes = cast(list[dict[str, Any]], topology.get("nodes", []))
+
+    for field in ("node_id", "mac", "name"):
+        for node in nodes:
+            value = node.get(field)
+            if value and str(value).strip().lower() == wanted:
+                return str(node.get("node_id"))
+
+    raise ResourceNotFoundError("topology_node", device_id)
```
[the remark this answers is not in the document]
Comment (jpennin5) [reply src/tools/topology.py]: Fixed in 0a022d8 — the error now reads `device '<id>' not found`, and the test asserts the wording.

**Comment by enuno** [comment]
### Summary
Request changes. The PR is not yet merge-ready because the current status checks are failing.

### Critical findings
- Merge state: \ / \
- Failing check(s): Mermaid Diagram Sync Assistant
- The changed topology surface should be revalidated after the failing check is resolved.

### Suggestions
1. Resolve the failing status check(s) and rerun CI.
2. Reconfirm the changed topology behavior against the intended controller/fixture shapes.
3. Ensure regression tests cover the original failure mode and edge cases introduced here.
4. Update docs if any request/response or error semantics changed.

### Verification checklist for the author
- [ ] Required checks are passing
- [ ] Regression tests cover the changed path
- [ ] Lint / format / type checks pass locally
- [ ] Documentation is updated where user-facing behavior changed
- [ ] No secrets, credentials, or controller-identifying data were introduced

**Review by enuno — CHANGES_REQUESTED** [review]
Merge blocked until the failing status checks are resolved and the topology changes are revalidated.

**Comment by jpennin5** [comment]
The failing check is the `Mermaid Diagram Sync Assistant` GitHub App, not a workflow in this repository. Its check-run output on this PR reads: "Webhook Processing Failed — An unexpected error occurred while processing the webhook. Please try pushing a new commit or check your repository configuration." That processing happens on the app's side; pushing new commits to these PRs has re-triggered the check with the same result, so it is not reachable from a fork.

This is the loop documented in #112: the app has errored on every fork PR since late June, fork CI runs sit at `action_required` awaiting maintainer approval, and the daily automated review then reads the app error as a failing required check and requests changes.

Every check that can conclude on a fork PR is green here (both Socket Security checks). The remaining steps are repo-side — approving the CI runs and either repairing/removing the app integration or unrequiring its check. #117 merged with the identical check state, so the merge path exists. Happy to rebase or adjust anything code-side the moment CI can actually run.

**Comment by enuno** [comment]
Summary: request changes

Critical findings:
- Merge conflict / dirty merge state blocks safe review.

Suggestions:
1. Rebase onto main and resolve conflicts in: API.md, CHANGELOG.md, src/tools/topology.py, tests/unit/tools/test_topology_tools.py.
2. Re-run the touched test suite and pre-commit.
3. Re-check the feature behavior called out in the PR description.

Verification checklist:
- [ ] Rebasing completed cleanly
- [ ] Tests pass
- [ ] Lint/type checks pass
- [ ] Documentation remains aligned with implementation

**Review by enuno — CHANGES_REQUESTED** [review]
Automated review: merge conflicts/dirtiness detected (CONFLICTING/DIRTY). Please rebase and resolve before resubmitting.

**Comment by jpennin5** [comment]
Rebased onto current `main` (ccd677b) — the conflict was in CHANGELOG, and it is resolved.

Resolution was gated rather than eyeballed: every CHANGELOG bullet already on `main` is verified still present (union, not overwrite), the sibling work merged into the shared files is verified intact, the full unit suite passes, and pre-commit passes for every file the PR touches.

CodeQL's dual-import finding is fixed: the module alias import is gone, the helpers use the local from-imports the rest of the file already used, and the two patches use string targets like every other suite here.

GitHub now reports this PR mergeable.

**Comment by enuno** [comment]
## Summary
Request changes.

## Critical findings
- CI is not green on this PR.
- Failing checks: Mermaid Diagram Sync Assistant; Pre-commit Hooks.
- Touched scope: API.md, CHANGELOG.md, src/tools, tests/unit.

## Suggestions
1. Fix the failing checks on the current branch and rerun them until the gate is green.
2. Re-verify the changed code paths and tests for the touched scope above.
3. If any failing security or lint checks are expected, document the rationale and follow-up needed.

## Verification checklist
- [ ] `pre-commit run --all-files` passes
- [ ] Security/lint checks are green or explicitly justified
- [ ] Relevant unit tests cover the changed behavior
- [ ] No unintended API or documentation regressions

**Review by enuno — CHANGES_REQUESTED** [review]
CI is red on this PR (Mermaid Diagram Sync Assistant; Pre-commit Hooks). Please resolve the failing checks before merge.

**Comment by enuno** [comment]
## Automated PR Review: #136

**Summary:** request changes

Fixes a real usability bug in topology lookups where legacy device IDs or MACs silently returned empty results. The new resolver accepts node id, MAC, or name and raises ResourceNotFoundError for unknown identifiers.

### Critical findings
- **Merge conflict:** GitHub reports mergeable: CONFLICTING / mergeStateStatus: DIRTY. This must be resolved before review can conclude.
- Pre-commit Hooks is failing due to repo-wide detect-secrets baseline drift, not this diff.
- The _resolve_topology_node function uses .strip().lower() for case-insensitive matching, which is reasonable but could cause collisions if two nodes have names that differ only by case/whitespace.

### Suggestions
1. Rebase on the current main and resolve conflicts.
2. Consider whether case-insensitive name matching could match the wrong device when names differ only by case; if so, document the behavior.
3. Verify API.md accurately describes the accepted identifier forms and the new error behavior.

### Verification checklist for the author
- [ ] Resolve merge conflicts on src/tools/topology.py and tests/unit/tools/test_topology_tools.py.
- [ ] Rebase after the detect-secrets baseline is fixed on main.
- [ ] Run the live Network 10.5.67 verification again after rebase.

**Review by enuno — CHANGES_REQUESTED** [review]
Automated review: this PR has merge conflicts (mergeable: CONFLICTING). Please rebase on the current main, resolve conflicts, and re-run tests.
